### PR TITLE
Add provisioner for Modal push work pools

### PR DESCRIPTION
**Pull request by desertaxle**
Adds a `Provisioner` implementation to configure a Modal push work pool via the Prefect CLI. The provisioner will handle creating a `ModalCredentials` block and configuring the created work pool to use the new block. The provisioner will prompt the user to install `modal` if it is not present and will prompt the user to generate a new token if one is not found in the current Modal configuration.

This PR adds Modal push work pool configuration information to the serverless push work pools guide.

### Example

Screenshot 2024-01-17 at 3 10 03 PM

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [x] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Review by jakekaplan — APPROVED** [review]
Looks great to me!

File: docs/guides/deployment/push-work-pools.md [hunk 1]
```diff
@@ -519,6 +550,14 @@ In the examples below, we'll create a push work pool via the Prefect Cloud UI.
 
     Once the Service account is created, navigate to its *Keys* page to add an API key. Create a JSON type key, download it, and store it somewhere safe for use in the next section.
 
+=== "Modal"
+
+    A Modal API token is required to push work to Modal.
+
+    Create a Modal API token by navigating to Settings in the Modal UI. In the API Tokens section of the Settings page, click *New Token*.
```
Comment (discdiver) [inline docs/guides/deployment/push-work-pools.md:557]:
```suggestion
    Create a Modal API token by navigating to **Settings** in the Modal UI. In the **API Tokens** section of the Settings page, click **New Token**.
```

File: docs/guides/deployment/push-work-pools.md [hunk 2]
```diff
@@ -394,6 +413,18 @@ Here's the command to create a new push work pool and configure the necessary in
...
+=== "Modal"
+    <div class="terminal">
+
+    ```bash
+    prefect work-pool create --type modal:push --provision-infra my-modal-pool 
+    ```
+
+    </div>
+
+    Using the `--provision-infra` flag will trigger the creation of a `ModalCredentials` block in your Prefect Cloud workspace. This block will store your Modal API token, which is used to authenticate with Modal's API. By default, the token for your current Modal profile will be used for the new `ModalCredentials` block. If we are unable to discover a Modal API token for your current profile, you will be prompted to create a new one.
```
Comment (discdiver) [inline docs/guides/deployment/push-work-pools.md:425]:
```suggestion
    Using the `--provision-infra` flag will trigger the creation of a `ModalCredentials` block in your Prefect Cloud workspace. This block will store your Modal API token, which is used to authenticate with Modal's API. By default, the token for your current Modal profile will be used for the new `ModalCredentials` block. If Prefect is unable to discover a Modal API token for your current profile, you will be prompted to create a new one.
```

**Review by discdiver — COMMENTED** [review]
Thank you! A few minor nits. Other than that, looks good for docs. 

Added a commit that adds Modal to the list of options at the top of the push work pools page.

**Review by billpalombi — APPROVED** [review]
Approving docs

**Pull request merged**